CIP-ledger-evolution/README.md

```diff
@@ -0,0 +1,302 @@
+
+#### Ledger events
+
+Some provenance about the ledger calculations is provided by ledger events.
+These events come with zero cost to a running node if not used and are not stored in the ledger state.
+
+### Soft forks and Hard forks
+
+We make the following definitions:
+
+**Hard fork** - A *hard fork* is a change to the protocol (not restricted to the ledger)
+resulting in a single new (CBOR) block being valid.
+
+**Soft fork** - A *soft fork* is a change to the protocol (not restricted to the ledger)
+resulting in fewer blocks being valid.
+A good way to think about soft forks is the property that a node which only validates
+(and does not produce blocks) does not need a software upgrade in order to continue following the block chain.
+Soft forks do not seem to come up often in practice.
+
+### Serialization
+
+Transactions and blocks are serialized with
+[CBOR](https://www.rfc-editor.org/rfc/rfc7049)
+and specified with
+[CDDL file](https://www.rfc-editor.org/rfc/rfc8610).
+
+The ledger state is also currently also serialized with CBOR,
+but this may change in the future.
+
+### Plutus script context
+
+The ledger and Plutus scripts have a common interface, namely the script context.
+See [CIP-35] for more information.
+Note that CIPs relating to the script context are relevant to both the ledger and to the Plutus CIP categories.
+
+## Specification
+
+This proposal deals only with the types of change listed in
+[Types of change](#types-of-change), all others are out of scope.
+
+### Changes that require a CIP
+
+This proposal recommends that some of the changes listed in "Types of change" (specified below) should:
+
+* Be proposed in a CIP.
+* Go through additional process in addition to the usual CIP process.
+
+The additional process mostly takes the form of additional information that should be present in
+the CIP before it moves to particular stages. As such, it is up to the CIP Editors to enforce this.
+
+The requirement to propose a change via a CIP is, as all CIPs are, advisory.
+In exceptional circumstances or where swift action is required, we expect that changes may still
+be made without following this process. In such circumstances,
+a retrospective CIP SHOULD be made after the fact to record the changes and the rationale for them.
+
+Changes that require a CIP do not have to each be in an individual CIP, they can be included in batches or in other CIPs.
+So, for example, a single CIP could propose multiple new related fields in the transaction.
+
+It is very likely the case that if a change does not require a soft or hard fork,
+it does not require a CIP, except in the case of bugs.
+
+A "bug fix" is a change to behavior where:
+- The implemented behavior does not match the specification; or
+- The specified behavior is clearly wrong (in the judgment of relevant experts)
+
+#### Changes that *do not* require a CIP
+
+* performance improvements
+* bug fixes
+* ledger events
```

Comment:
I'm open to including them! I was mostly just trying to find the right place to draw the fence, and I went with "things that change the semantics are inside the fence, and things that do not can just be raised as normal GitHub issues). The stakes (pun always intended!) are low if the semantics are not changing.

But, to argue with myself, if, say, a Rust node were to be written, I would want the events to be uniform. I think that convinces me.

Maybe the fence could be "the things that every implementation should be consistent about"?

Reply:
That's the kind of things I am thinking of, eg. interoperability and heterogeneity of participants in the network. And yes, I agree with your last definition of the fence @JaredCorduan. 
So regarding events: Either they are an implementation details and each node/eco-system will have their own definitions, in which case they must be left out of the CIP and I would even not mention them as this could mislead people ; or they are an integral part of the ledger _specification_ and an interface to the outside world and they should be included.
To give a concrete analogy drawn from my personal experience: In the Hydra team, we consider the logs to be part of the interface of the Hydra node (because they are critical for operating the system) hence we provide a "specification" for them (in this case just a schema). Other teams/systems might have different says on this.

Reply:
ok, great. I love the idea of making this fence explicit somewhere in the document, and I think I like this new definition. But of course, I'd be even happier if I heard Michael's counter-thoughts. :)

Reply:
I think Arnaud and I agree on the principles, I just think that the ledger events are "outside the fence". I think it would be perfectly reasonable for a competing implementation of Cardano to take its own, totally different approach to observability of what's going on in the ledger.

Reply:
We are definitely on the same page 👍